### PR TITLE
Fix a bug in MappingQGeneric::reinit().

### LINKED ISSUE #2661: Vertex position update leaves FEValues stale
Reference to this [forum post](https://groups.google.com/forum/#!topic/dealii/c8QfaMThBaQ):

When moving the mesh as is done in step-18, at least 1 global refinement must be performed in order for the geometry update to be propagated to any DoFHandlers that are attached to the triangulation. An MWE is located here: [mesh_movement.cc.zip](https://github.com/dealii/dealii/files/292744/mesh_movement.cc.zip)

### PR DESCRIPTION
**Pull request by bangerth**
Specifically, we thought that we were on the same cell as the last call to
FEValues::reinit() and so did not re-compute information -- but the vertices
of the triangulation had been moved in between, and consequently the previously
computed data was stale.

This fixes #2661.

### CODE REVIEW

File: doc/news/changes.h [hunk 1]
```diff
@@ -336,6 +336,16 @@ inconvenience this causes.
...
 
 <ol>
+ <li> Fixed: FEValues::reinit() would sometimes try to be overly
+ clever and not re-compute information when called with the same
+ cell twice in a row, even if the underlying triangulation had
+ been moved, translated, stretched, or otherwise had its vertex
+ locations changed between the two calls to FEValues::reinit().
+ This is now fixed.
+ <br>
+ (Wolfgang Bangerth, Rajat Arora 2016/07/11)
```
Comment (jppelteret) [inline doc/news/changes.h]: No love for my MWE? ;-)

Comment (bangerth) [inline doc/news/changes.h]: Oh, my sincere apologies. I did not sufficiently investigate who that test came from originally. Will be added immediately!

Comment (bangerth) [inline doc/news/changes.h]: See #2806.

Comment (jppelteret) [inline doc/news/changes.h]: Its no problem! Its just that my funding grants are evaluated solely on the number of internet points that I have, so every bit counts :-P

Comment (bangerth) [inline doc/news/changes.h]: We look out for each other over here in this project, so I will gladly make sure you get more points ;-)

**Pull request merged**